src/kernel/arch/aarch64/boot.zig

```diff
@@ -8,32 +9,39 @@ extern var KERNEL_STACK_END: *usize;
 
 export fn _start() linksection(".text.boot") callconv(.Naked) noreturn {
     // Halt all cores other than the primary core, until the kernel has multicore support
-    const core_id = asm ("mrs %[res], mpidr_el1"
-        : [res] "=r" (-> usize)
-    ) & 3;
-    if (core_id != 0) {
-        while (true) {
-            asm volatile ("wfe");
-        }
-    }
+    asm volatile (
+        \\ mrs x0, mpidr_el1
+        \\ mov x1, #3
+        \\ ands x0, x0, x1
+        \\ beq cpu0
+        \\hang:
+        \\ wfe
+        \\ b hang
+        \\cpu0:
+    );
 
     // Setup the stack
-    asm volatile ("mov sp, %[stack_end]"
+    asm volatile (
+        \\mov sp, %[stack_end]
         :
         : [stack_end] "r" (@ptrCast([*]u8, &KERNEL_STACK_END))
     );
 
+    interrupts.exception_handler_depth = 0;
+
     // Setup the exception table
     asm volatile (
-        \\msr vbar_el1, %[table_addr]
-        \\msr vbar_el2, %[table_addr]
-        \\msr vbar_el3, %[table_addr]
+        \\ msr vbar_el1, %[table_addr]
+        \\ msr vbar_el2, %[table_addr]
+        \\ msr vbar_el3, %[table_addr]
         :
         : [table_addr] "r" (@ptrToInt(&interrupts.exception_table))
     );
 
+    arch.enableFlatMmu();
+
     // The rpi puts the board part number in midr_el1
-    const board = detectBoard();
+    const board = rpi.RaspberryPiBoard.RaspberryPi3; // detectBoard();
```

Comment:
Why not `detectBoard()`?

Reply:
I suspect a compiler defect when returning an optional struct from a function. I intend to reduce and isolate this when I am at my lab today.

Reply:
Nice!